presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java

```diff
@@ -201,8 +202,10 @@ public HivePageSink(
         ImmutableList.Builder<DataColumn> dataColumns = ImmutableList.builder();
         for (HiveColumnHandle column : inputColumns) {
             if (column.isPartitionKey()) {
-                partitionColumnNames.add(column.getName());
-                partitionColumnTypes.add(typeManager.getType(column.getTypeSignature()));
+                if (!isHiddenColumnHandle(column)) {
```

Comment:
You shouldn't have put the hidden columns in the list of `columnHandle`s when you constructed the list for `beginInsert` and `beginCreateTable`.